src/mip/mip_logger.c

```diff
@@ -0,0 +1,55 @@
+
+#include "mip_logger.h"
+
+#include <stddef.h>
+
+////////////////////////////////////////////////////////////////////////////////
+///@brief Global logging callback. Do not access directly
+mip_logger_callback _mip_logger_callback = NULL;
+
+////////////////////////////////////////////////////////////////////////////////
+///@brief Global logging level. Do not access directly
+mip_logger_level _mip_logger_level = MIP_LOGGER_LEVEL_OFF;
+
+////////////////////////////////////////////////////////////////////////////////
+///@brief Global logging user data. Do not access directly
+void* _mip_logger_user_data = NULL;
+
+////////////////////////////////////////////////////////////////////////////////
+///@brief Initializes the logger with a callback and user data
+///
+///@param callback The callback to execute when there is data to log
+///@param level    The level that the MIP SDK should log at
+///@param user     User data that will be passed to the callback every time it is excuted
+///
+void mip_logger_init(mip_logger_callback callback, mip_logger_level level, void* user)
+{
+  _mip_logger_callback = callback;
+  _mip_logger_level = level;
+  _mip_logger_user_data = user;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+///@brief Gets the currently active logger callback
+///
+///@return The currently active logger callback
+///
+mip_logger_callback mip_logger_get_callback()
```

Comment:
The other MIP functions don't include 'get' in the name, I suggest the same here for consistency (unless we decide to change the other functions).